build-system/tasks/performance-test/compile-scripts.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const argv = require('minimist')(process.argv.slice(2), {
+  default: {compile: true},
+});
+const {dist} = require('../dist');
+const {EXPERIMENT, urlToCachePath} = require('./helpers');
+
+/**
+ * Compiles and minifies AMP runtime and components required by the urls
+ * argument, unless the --no-compile flag is passed.
+ * @param {Array<string>} urls
+ */
+async function compileScripts(urls) {
+  if (argv.compile) {
+    process.argv.push(
+      `--extensions_from=${urls
+        .map(url => urlToCachePath(url, EXPERIMENT))
+        .join(',')}`
+    );
```

Comment:
It's a bad idea to modify `process.argv` because `argv` will no longer be accurate inside other modules that may have consumed `process.argv` at the start. Also, by pushing an argument, you're potentially introducing duplicates that only `compile-scripts.js` will see, but other JS modules will not.

Instead of this, you could add a method to `extension-helpers.js` that overrides the set of extensions to build. Happy to chat offline about what this could look like.